portal/import_template.php

```diff
@@ -12,106 +12,227 @@
-        redirect();
-    } elseif (realpath($rebuiltPath) != realpath($filename)) { // these need to match to be valid request
-        redirect();
-    } elseif (stripos(realpath($filename), realpath($knownPath)) === false) { // this needs to pass be a valid request
-        redirect();
+if ($_REQUEST['mode'] === 'editor_render_html') {
+    if ($_REQUEST['docid']) {
+        $content = $templateService->fetchTemplate($_REQUEST['docid']);
+        $template_content = $content['template_content'];
+        if ($content['mime'] == 'application/pdf') {
+            //$template_content = 'data:application/pdf;base64,' . base64_encode($template_content);
+            $content = "<iframe width='100%' height='100%' src='data:application/pdf;base64, " .
+                attr(base64_encode($template_content)) . "'></iframe>";
+            echo $content;
+            exit;
+        }
+        renderEditorHtml($_REQUEST['docid'], $template_content);
+    } else {
+        die(xlt('Invalid File'));
     }
-
-    return $rebuiltPath;
+} elseif (!empty($_GET['templateHtml'] ?? null)) {
+    renderEditorHtml($_REQUEST['docid'], $_GET['templateHtml']);
 }
 
-function redirect()
+function renderEditorHtml($template_id, $content)
 {
-    header('HTTP/1.0 404 Not Found');
-    die();
-}
+    $lists = [
+        '{ParseAsHTML}', '{TextInput}', '{sizedTextInput:120px}', '{smTextInput}', '{TextBox:03x080}', '{CheckMark}', '{ynRadioGroup}', '{TrueFalseRadioGroup}', '{DatePicker}', '{DateTimePicker}', '{StandardDatePicker}', '{CurrentDate:"global"}', '{CurrentTime}', '{DOS}', '{ReferringDOC}', '{PatientID}', '{PatientName}', '{PatientSex}', '{PatientDOB}', '{PatientPhone}', '{Address}', '{City}', '{State}', '{Zip}', '{PatientSignature}', '{AdminSignature}', '{AcknowledgePdf: : }', '{EncounterForm:LBF}', '{Medications}', '{ProblemList}', '{Allergies}', '{ChiefComplaint}', '{DEM: }', '{HIS: }', '{LBF: }', '{GRP}{/GRP}'
+    ];
+    ?>
+    <!DOCTYPE html>
+    <html>
+    <head>
+        <?php Header::setupHeader(['ckeditor']); ?>
+    </head>
+    <style>
+      input:focus,
+      input:active
+      {
+        outline:0px !important;
+        -webkit-appearance:none;
+        box-shadow: none !important;
+      }
+      .list-group-item {
+        font-size: .9rem;
+      }
+    </style>
+    <body>
+        <div class="container-fluid mt-2">
+            <div class="row">
+                <div class="col-10 px-1">
+                    <form class="sticky-top" action='./import_template.php' method='post'>
+                        <input type="hidden" name="docid" value="<?php echo $template_id ?>">
+                        <input type='hidden' name='mode' value="save">
+                        <input type='hidden' name='service' value='window'>
+                        <textarea class='inline-editor' contenteditable='true' cols='80' rows='10' id='templateContent' name='content'><?php echo $content ?></textarea>
```

Comment:
text() if $content does not contain html

Reply:
Yeah this is a tough one. Tipplers can be all text or all HTML or both. I'll have to figure it out